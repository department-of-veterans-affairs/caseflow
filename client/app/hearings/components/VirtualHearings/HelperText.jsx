import React from 'react';
import PropTypes from 'prop-types';

export const HelperText = ({ label }) => (
  <div className="helper-text" >
    {label}
  </div>
);

HelperText.propTypes = {
  label: PropTypes.string,
};
