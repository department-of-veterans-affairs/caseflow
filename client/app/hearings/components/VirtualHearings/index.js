export * from '../details/HearingEmail';
export * from './Section';
export * from './Timezone';
export * from './HelperText';
