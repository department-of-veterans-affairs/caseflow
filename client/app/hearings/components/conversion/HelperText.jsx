import React from 'react';

import { helperLabel } from '../details/style';

export const HelperText = ({ label }) => (
  <span {...helperLabel}>
    {label}
  </span>
);
