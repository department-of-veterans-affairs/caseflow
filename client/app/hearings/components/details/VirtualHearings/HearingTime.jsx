import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'glamor';

import { hearingTimeOptsWithZone } from '../../../utils';
import SearchableDropdown from '../../../../components/SearchableDropdown';

export const HearingTime = ({ readOnly, value, errorMessage, onChange }) => {
  return (
    <SearchableDropdown
      readOnly={readOnly}
      name="Hearing Time"
      placeholder="Select a time"
      options={hearingTimeOptsWithZone()}
      value={value}
      onChange={(option) => onChange(option ? option.value : null)}
      errorMessage={errorMessage}
      strongLabel
    />
  );
};

HearingTime.defaultProps = {
  value: null,
  readOnly: false
};

HearingTime.propTypes = {
  errorMessage: PropTypes.string,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  value: PropTypes.string
};
