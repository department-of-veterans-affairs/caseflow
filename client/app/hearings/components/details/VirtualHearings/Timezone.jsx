import React from 'react';
import PropTypes from 'prop-types';

import SearchableDropdown from '../../../../components/SearchableDropdown';
import { timezones } from '../../../utils';
import { timezoneDropdownStyles, timezoneStyles } from '../style';

export const Timezone = ({ readOnly, value, errorMessage, onChange, time }) => {
  const { options, commonsCount } = timezones(time);

  return (
    <SearchableDropdown
      styling={timezoneStyles(commonsCount)}
      dropdownStyling={timezoneDropdownStyles(commonsCount)}
      name="Timezone"
      readOnly={readOnly}
      placeholder="Select a timezone"
      options={options}
      value={value}
      onChange={(option) => onChange(option ? option.value : null)}
      errorMessage={errorMessage}
      strongLabel
    />
  );
};

Timezone.defaultProps = {
  value: null,
  readOnly: false
};

Timezone.propTypes = {
  errorMessage: PropTypes.string,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
  value: PropTypes.string,
  time: PropTypes.string
};
