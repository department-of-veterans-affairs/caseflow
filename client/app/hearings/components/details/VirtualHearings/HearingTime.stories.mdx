import { useState } from "react";
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { action } from "@storybook/addon-actions";
import { select } from "@storybook/addon-knobs";

import { HearingTime } from "./HearingTime";
import { hearingTimeOptsWithZone } from "../../../utils";

export const defaultProps = {};

<Meta
  title="Hearings/Components/Hearing Details/Virtual Hearings/HearingTime"
  component={HearingTime}
/>

# Hearing Time

### Default

<Preview>
  <Story name="Default">
    {() => {
      const [value, setValue] = useState("");
      return (
        <div style={{ width: 500 }}>
          <HearingTime value={value} onChange={setValue} />
        </div>
      );
    }}
  </Story>
</Preview>

### Disabled

<Preview>
  <Story name="Disabled">
    {() => {
      const [value, setValue] = useState("");
      return (
        <div style={{ width: 500 }}>
          <HearingTime
            readOnly
            value={select("Time", hearingTimeOptsWithZone(), "08:30")}
            onChange={setValue}
          />
        </div>
      );
    }}
  </Story>
</Preview>

### With Errors

<Preview>
  <Story name="With Errors">
    {() => {
      const [value, setValue] = useState("");
      return (
        <div style={{ width: 500 }}>
          <HearingTime
            errorMessage="Invalid time selected"
            value={value}
            onChange={setValue}
          />
        </div>
      );
    }}
  </Story>
</Preview>

### With Pre-Selected Option

<Preview>
  <Story name="With Pre-Selected Option">
    {() => {
      const [value, setValue] = useState("08:30");
      return (
        <div style={{ width: 500 }}>
          <HearingTime value={value} onChange={setValue} />
        </div>
      );
    }}
  </Story>
</Preview>

<Props of={HearingTime} />
