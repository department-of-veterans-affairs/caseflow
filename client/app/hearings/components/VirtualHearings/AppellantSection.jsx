import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { sprintf } from 'sprintf-js';

import COPY from '../../../../COPY';
import Alert from '../../../components/Alert';
import { AddressLine } from '../details/Address';
import { VirtualHearingSection } from './Section';
import { HelperText } from './HelperText';
import { VirtualHearingEmail } from './Emails';
import { Timezone } from './Timezone';
import { marginTop, noMaxWidth } from '../details/style';
import { ReadOnly } from '../details/ReadOnly';

export const AppellantSection = ({
  hearing,
  virtualHearing,
  errors,
  type,
  virtual,
  video,
  readOnly,
  showDivider,
  update,
  appellantTitle,
  showOnlyAppellantName,
  showMissingEmailAlert
}) => {
  // Depending on where this component is used, the *FullName fields will be available.
  // If they aren't, the *FirstName/*LastName fields should be available.
  const appellantName = hearing?.appellantIsNotVeteran ?
    (hearing?.appellantFullName || `${hearing?.appellantFirstName} ${hearing?.appellantLastName}`) :
    (hearing?.veteranFullName || `${hearing?.veteranFirstName} ${hearing?.veteranLastName}`);
  const showTimezoneField = virtual && !video;

  // determine whether to show a missing email underneath readonly email
  const showMissingAlert = readOnly && showMissingEmailAlert && !virtualHearing?.appellantEmail;

  return (
    <VirtualHearingSection label={appellantTitle} showDivider={showDivider}>
      {/*
        * Appellant Name and Address
        */}
      {showOnlyAppellantName ? (
        <ReadOnly
          label={`${appellantTitle} Name`}
          text={appellantName}
        />
      ) :
        (
          <AddressLine
            name={appellantName}
            addressLine1={hearing?.appellantAddressLine1}
            addressState={hearing?.appellantState}
            addressCity={hearing?.appellantCity}
            addressZip={hearing?.appellantZip}
          />
        )}
      {/*
        * Timezone fields
        */}
      {showTimezoneField && (
        <div className={classNames('usa-grid', { [marginTop(30)]: true })}>
          <div className={classNames('usa-width-one-half', { [noMaxWidth]: true })} >
            <Timezone
              required
              value={virtualHearing?.appellantTz}
              onChange={(appellantTz) => update('virtualHearing', { appellantTz })}
              time={hearing.scheduledTimeString}
              label={`${appellantTitle} Timezone`}
              name="appellantTz"
              errorMessage={errors?.appellantTz}
            />
            <HelperText label={COPY.VIRTUAL_HEARING_TIMEZONE_HELPER_TEXT} />
          </div>
        </div>
      )}
      {/*
        * Email fields
        */}
      <div id="email-section" className={classNames('usa-grid', { [marginTop(30)]: true })}>
        <div className={classNames('usa-width-one-half', { [noMaxWidth]: true })} >
          <VirtualHearingEmail
            required
            readOnly={readOnly}
            label={`${appellantTitle} Email`}
            emailType="appellantEmail"
            email={virtualHearing?.appellantEmail}
            error={errors?.appellantEmail}
            type={type}
            update={update}
          />
          {showMissingAlert && (
            <div>
              <Alert
                message={sprintf(COPY.MISSING_EMAIL_ALERT_MESSAGE, appellantTitle)}
                type="info"
                scrollOnAlert={false}
              />
            </div>
          )}
        </div>
      </div>
    </VirtualHearingSection>
  );
};

AppellantSection.propTypes = {
  hearing: PropTypes.object,
  virtualHearing: PropTypes.object,
  errors: PropTypes.object,
  type: PropTypes.string,
  update: PropTypes.func,
  virtual: PropTypes.bool,
  video: PropTypes.bool,
  readOnly: PropTypes.bool,
  appellantTitle: PropTypes.string,
  showOnlyAppellantName: PropTypes.bool,
  showDivider: PropTypes.bool,
  showMissingEmailAlert: PropTypes.bool
};
