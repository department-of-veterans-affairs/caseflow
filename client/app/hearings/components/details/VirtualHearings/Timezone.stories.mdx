import { useState } from "react";
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { action } from "@storybook/addon-actions";
import { select } from "@storybook/addon-knobs";

import { Timezone } from "./Timezone";
import { timezones } from "../../../utils";

export const defaultProps = {};

<Meta
  title="Hearings/Components/Hearing Details/Virtual Hearings/Timezone"
  component={Timezone}
/>

# Default

<Preview>
  <Story name="Default">
    {() => {
      const [value, setValue] = useState("");
      return (
        <div style={{ width: 300 }}>
          <Timezone value={value} onChange={setValue} />
        </div>
      );
    }}
  </Story>
</Preview>

<Props of={Timezone} />
