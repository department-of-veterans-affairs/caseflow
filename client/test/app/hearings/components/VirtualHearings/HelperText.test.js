import React from 'react';
import { shallow } from 'enzyme';

import { HelperText } from 'app/hearings/components/VirtualHearings/HelperText';

const label = 'Something helpful';

describe('HelperText', () => {
  test('Matches snapshot with default props', () => {
    // Run the test
    const helperText = shallow(
      <HelperText label={label} />
    );

    // Assertions
    expect(helperText.text()).toEqual(label);
    expect(helperText).toMatchSnapshot();

  });
})
;
