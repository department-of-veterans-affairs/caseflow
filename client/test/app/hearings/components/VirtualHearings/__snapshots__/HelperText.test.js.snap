// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperText Matches snapshot with default props 1`] = `
<div
  className="helper-text"
>
  Something helpful
</div>
`;
